### PR TITLE
support for v1.Event

### LINKED ISSUE #83: support for v1.Event
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE
**What happened**:
Currently upgrade-manager do not emit `v1.Event`. 
**What you expected to happen**:
Emit `v1.Event` when a Rolling Upgrade has Started, Finished (success or failure) etc

### PR DESCRIPTION
**Pull request by vigith**
fixes #83

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -24,6 +24,7 @@ require (
 	k8s.io/client-go v0.17.0
 	k8s.io/utils v0.0.0-20191218082557-f07c713de883 // indirect
 	sigs.k8s.io/controller-runtime v0.4.0
+	sigs.k8s.io/controller-tools v0.2.4 // indirect
```
Comment (eytan-avisror) [inline go.mod:27]: is there a reason we're adding this?

File: controllers/rollingupgrade_controller.go [hunk 2]
```diff
@@ -872,25 +892,38 @@ func (r *RollingUpgradeReconciler) UpdateInstances(ctx *context.Context,
...
+		})
 		go r.UpdateInstance(ctx, ruObj, instance, launchDefinition, KubeCtlCall, ch)
 	}
 
 	// wait for upgrades to complete
 	nodesProcessed := 0
 	var instanceUpdateErrors []error
-Loop:
+
 	for err := range ch {
```
Comment (eytan-avisror) [inline controllers/rollingupgrade_controller.go:909]: wont this get stuck until channel is closed?

Comment (vigith) [reply controllers/rollingupgrade_controller.go:909]: `if nodesProcessed == totalNodes { break }` should break the loop if we have `nodesProcessed == totalNodes`

File: go.mod (hunk 1 above)
Comment (vigith) [reply go.mod]: removed

**Pull request merged**